### PR TITLE
Adds Bulk Check-in under the Users view '#assets' tab

### PR DESCRIPTION
**Pull request by Godmartinz**
# Description
This adds an option to the bulk actions menu to check in multiple Assets associated with the current user. This option is only available under the user view #assets tab currently.
image
image

A success message representing the number of items that were checked in will displayed back at the users view blade.
image

Fixes #SC-10640

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* PHP version:
* MySQL version
* Webserver version
* OS version

# Checklist:

- [ ] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [ ] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by snipe** [comment]
@Godmartinz ignore my comments in Slack - I thought you were referring to something else. Can we fix the wee-ness of that menu? (I know that's not something you regressed here, it's like that on master, but it looks weird.)

File: resources/views/users/view.blade.php [hunk 1]
```diff
@@ -670,7 +669,7 @@
                     data-sort-order="asc"
                     data-sort-name="name"
                     data-toolbar="#assetsBulkEditToolbar"
-                    data-bulk-button-id="#bulkAssetEditButton"
+                    data-bulk-button-id="#bulkAssetEditButton, #checkInAll"
```
Comment (snipe) [inline resources/views/users/view.blade.php]: I don't think elements can have more than one ID

File: resources/lang/en/admin/hardware/form.php [hunk 2]
```diff
@@ -8,6 +8,7 @@
   'bulk_delete_warn'	=> 'You are about to delete :asset_count assets.',
   'bulk_restore_warn'	=> 'You are about to restore :asset_count assets.',
     'bulk_update'		=> 'Bulk Update Assets',
+    'bulk_checkin'      => 'Bulk Checkin Assets',
```
Comment (snipe) [inline resources/lang/en/admin/hardware/form.php]: We can probably re-use the "Checkin Assets" or "Checkin" string here - it's a bulk menu, so I think the bulk part is implied.

File: resources/views/partials/asset-bulk-actions.blade.php [hunk 3]
```diff
@@ -19,8 +19,13 @@
         @endcan
         @else
         @can('update', \App\Models\Asset::class)
-            <option value="edit">{{ trans('button.edit') }}</option>
```
Comment (snipe) [inline resources/views/partials/asset-bulk-actions.blade.php:22]: Small thing, but can you revert the change to the cuddling here? We always want a space between `{{` and `}}` and whatever is inside.

File: app/Http/Controllers/Assets/BulkAssetsController.php [hunk 4]
```diff
@@ -393,19 +407,116 @@ public function storeCheckout(AssetCheckoutRequest $request)
...
+                $asset = Asset::withTrashed()->find($assetId);
+                $asset->restore();
+            }
+            return redirect()->route('hardware.index')->with('success', trans('admin/hardware/message.restore.success'));
+        }
+    }
+
+    public function bulkCheckin(Request $request, $backto = null)
+    {
+        $assetIds =$request->get('ids');
```
Comment (spencerrlongg) [inline app/Http/Controllers/Assets/BulkAssetsController.php]: Nitpick: add space

File: app/Http/Controllers/Assets/BulkAssetsController.php [hunk 5]
```diff
@@ -393,19 +407,116 @@ public function storeCheckout(AssetCheckoutRequest $request)
...
+        $count = count($assetIds);
+        $user = '';
+        foreach($assetIds as $assetId) {
```
Comment (spencerrlongg) [inline app/Http/Controllers/Assets/BulkAssetsController.php:434]: Nitpick: add space

File: resources/views/hardware/bulk-checkin.blade.php [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+                    <!-- Status -->
+                    <div class="form-group {{ $errors->has('status_id') ? 'error' : '' }}">
+                      {{ Form::label('status_id', trans('admin/hardware/form.status'), array('class' => 'col-md-3 control-label')) }}
+                      <div class="col-md-7 required">
+                        {{ Form::select('status_id', $statusLabel_list, '', array('class'=>'select2', 'style'=>'width:100%','id' =>'modal-statuslabel_types', 'aria-label'=>'status_id')) }}
+                        {!! $errors->first('status_id', '<span class="alert-msg" aria-hidden="true"><i class="fas fa-times" aria-hidden="true"></i> :message</span>') !!}
+                      </div>
+                    </div>
+
+                  @include ('partials.forms.edit.location-select', ['translated_name' => trans('general.location'), 'fieldname' => 'location_id', 'help_text' => ($user->userloc->name) ? 'You can choose to check these assets in to a location other than the users location of '.$user->userloc->name.' if one is set.' : null])
```
Comment (spencerrlongg) [inline resources/views/hardware/bulk-checkin.blade.php]: I'm seeing a couple of problems here: 
1. What happens if the user doesn't have a location? Need some kind of null handler
2. Seems like maybe `$user` doesn't exist? I'm getting "Attempt to read property "name" on null" even when there is a location assigned - haven't done any deeper troubleshooting yet so it could be something else
   https://flareapp.io/share/OmVqOnR5
3. String needs to be translated

File: app/Http/Controllers/Assets/BulkAssetsController.php [hunk 7]
```diff
@@ -393,19 +407,116 @@ public function storeCheckout(AssetCheckoutRequest $request)
...
+                $asset->status_id = e($request->get('status_id'));
+            }
+
+            // This is just meant to correct legacy issues where some user data would have 0
+            // as a location ID, which isn't valid. Later versions of Snipe-IT have stricter validation
+            // rules, so it's necessary to fix this for long-time users. It's kinda gross, but will help
+            // people (and their data) in the long run
+
+            if ($asset->rtd_location_id == '0') {
+                \Log::debug('Manually override the RTD location IDs');
```
Comment (spencerrlongg) [inline app/Http/Controllers/Assets/BulkAssetsController.php]: Kind of a nitpick, but I think we'd rather do an import here, IDE handles that better (this looks like it's been copied from somewhere else)

**Review by spencerrlongg — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/Assets/BulkAssetsController.php [hunk 8]
```diff
@@ -15,6 +17,8 @@
 use Illuminate\Support\Facades\Session;
 use App\Http\Requests\AssetCheckoutRequest;
 use App\Models\CustomField;
+use Illuminate\Database\Eloquent\Builder;
+use PHPStan\BetterReflection\Reflection\Adapter\FakeReflectionAttribute;
```
Comment (spencerrlongg) [inline app/Http/Controllers/Assets/BulkAssetsController.php]: This looks like it was an accidental import

Comment (Godmartinz) [reply app/Http/Controllers/Assets/BulkAssetsController.php]: Yep, thanks.

File: resources/views/hardware/bulk-checkin.blade.php (hunk 6 above)
Comment (Godmartinz) [reply resources/views/hardware/bulk-checkin.blade.php]: The ternary takes care of that here no? Will get a translation in 👍

Comment (spencerrlongg) [reply resources/views/hardware/bulk-checkin.blade.php]: Oh, I missed that this was a ternary earlier. Maybe check like `$user->userloc ? ... : null` to fix?

File: resources/views/users/view.blade.php (hunk 1 above)
Comment (Godmartinz) [reply resources/views/users/view.blade.php]: This actually worked, but it is totally unnecessary now. I was messing around with a button when I passed this id and to my delight it worked 🙂 

removed

File: resources/views/hardware/bulk-checkin.blade.php (hunk 6 above)
Comment (Godmartinz) [reply resources/views/hardware/bulk-checkin.blade.php]: oh ok so you are seeing an issue--that was unclear to me. I will check on it first thing tomorrow 👍

File: resources/views/hardware/bulk-checkin.blade.php [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+    }
+  </style>
+
+
+  <div class="row">
+    <!-- left column -->
+    <div class="col-md-9">
+      <div class="box box-default">
+        <div class="box-header with-border">
+          <h2 class="box-title">{{trans('admin/hardware/form.bulk_checkin')}}</h2>
```
Comment (marcusmoore) [inline resources/views/hardware/bulk-checkin.blade.php:24]: This translation string got removed I think (it shows up `asset-bulk-actions.blade.php` as well

CleanShot 2023-09-20 at 11 40 38@2x

File: resources/views/hardware/bulk-checkin.blade.php [hunk 10]
```diff
@@ -0,0 +1,115 @@
+@extends('layouts/default')
+
+ Page title
```
Comment (marcusmoore) [inline resources/views/hardware/bulk-checkin.blade.php:3]: We should remove this line right?

CleanShot 2023-09-20 at 11 40 38@2x

File: resources/views/hardware/bulk-checkin.blade.php [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+@section('title')
+  {{ trans('admin/hardware/general.checkin') }}
+  @parent
+@stop
+
+ Page content
```
Comment (marcusmoore) [inline resources/views/hardware/bulk-checkin.blade.php:9]: We should remove this line right?

CleanShot 2023-09-20 at 11 40 38@2x

**Review by marcusmoore — CHANGES_REQUESTED** [review]
Leaning on @spencerrlongg for a thorough review but I noticed a couple visual fixes.

**Review by spencerrlongg — APPROVED** [review]
Looks good to me outside of the visual fixes Marcus noted.
